vscode/webviews/chat/Transcript.tsx

```diff
@@ -88,6 +89,9 @@ export const Transcript: FC<TranscriptProps> = props => {
 
     const lastHumanEditorRef = useRef<PromptEditorRefAPI | null>(null)
 
+    useCopyHandler()
+    useRegisterCopyHandler(copyButtonOnSubmit)
```

Comment:
```suggestion
    useEffect(() => {
        const handleCopyEvent = (event: ClipboardEvent) => {
            try {
                const selectedText = window.getSelection()?.toString() || ''
                if (!selectedText) return

                for (const handler of handlersRef.current) {
                    try {
                        handler(selectedText, 'Keydown')
                    } catch (error) {
                        console.error('Error in copy handler:', error)
                    }
                }
                getVSCodeAPI().postMessage({
                    command: 'copy',
                    text: selectedText,
                    eventType: 'Keydown',
                })
            } catch (error) {
                console.error('Error handling copy event:', error)
            }
        }

        document.addEventListener('copy', handleCopyEvent)
        return () => {
            document.removeEventListener('copy', handleCopyEvent)
        }
    }, [])
```

i thiink you can move your useEffect code from vscode/webviews/utils/CopyHandlerContext.tsx then we won't need it as a context provider